File: java/src/test/java/problems301_400/IsSubsequenceTest.java
package problems301_400;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class IsSubsequenceTest extends SampledTest {
    @Test
    public void test() {
        for (IsSubsequence solution : IsSubsequence.values()) {
            for (IsSubsequenceSample sample : getSamples(IsSubsequenceSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isSubsequence(sample.input.s, sample.input.t))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems301_400/IsSubsequenceTest.kt
package problems301_400

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class IsSubsequenceTest : SampledTest() {
    @Test
    fun test() =
        IsSubsequence.entries.forEach { solution ->
            getSamples<Array<IsSubsequenceSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.isSubsequence(input.s, input.t))
                    .isEqualTo(output)
            }
        }
}
